### PR TITLE
Feat: export dq only

### PR DESCRIPTION
**Pull request by Giuseppe5**
Supersedes #94  

The idea is to export only Integer weights + DQ. For this, we need to use PyTorch 2.2+ because of a bug in how constant values are handled at export time.

### CODE REVIEW

**Review by dineshchitlangia — APPROVED** [review]
Overall, LGTM.
Minor question pls:
` export_manager.change_weight_export(export_weight_q_node=True)`
In your change, how are we accounting for this weight export?

**Comment by Giuseppe5** [comment]
Last time I discussed with @fxmarty, he mentioned that it would be always preferable to export only Integer Weights -> DQ rather than Float Weights -> Q -> DQ, so this last option has been completely removed in this PR.

This is also related to the changes in #82 , where we assume that the weights have been exported without the Q node.

**Comment by dineshchitlangia** [comment]
> Last time I discussed with @fxmarty, he mentioned that it would be always preferable to export only Integer Weights -> DQ rather than Float Weights -> Q -> DQ, so this last option has been completely removed in this PR.
> 
> This is also related to the changes in #82 , where we assume that the weights have been exported without the Q node.

Thanks for the clarity @Giuseppe5 

I do not have write privileges to merge your changes so you will have to wait a bit more until someone can merge it.

File: examples/quantization/brevitas/quantize_llm.py [hunk 1]
```diff
@@ -1,13 +1,9 @@
...
 
-import torch
-from brevitas.export.onnx.standard.qcdq.manager import StdQCDQONNXManager
-from brevitas_examples.llm.llm_quant.export import brevitas_proxy_export_mode
-
 from optimum.amd import BrevitasQuantizationConfig, BrevitasQuantizer
 from optimum.amd.brevitas.accelerate_utils import calc_cpu_device_map, calc_gpu_device_map, offload_model, remove_hooks
 from optimum.amd.brevitas.data_utils import compute_perplexity, get_dataset_for_model
-from optimum.exporters.onnx import onnx_export_from_model
+from optimum.amd.brevitas.export import export_quantized_model
```
Comment (mht-sharma) [inline examples/quantization/brevitas/quantize_llm.py:6]:
```suggestion
from optimum.amd.brevitas.export import export_to_onnx
```
Can we keep the ONNX word in the loop to make it explicit. Other name suggestions, `quantized_model_to_onnx` or `save_quantized_model_as_onnx`

File: optimum/amd/brevitas/export.py [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import torch
+from brevitas.export.onnx.standard.qcdq.manager import StdQCDQONNXManager
+from brevitas_examples.llm.llm_quant.export import brevitas_proxy_export_mode
+
+from optimum.exporters.onnx import onnx_export_from_model
+
+
+def export_quantized_model(quantized_model, path, task="text-generation-with-past"):
```
Comment (mht-sharma) [inline optimum/amd/brevitas/export.py:8]: For this function lets match the necessary input signatures from here:  https://github.com/huggingface/optimum/blob/main/optimum/exporters/onnx/convert.py#L906

**Review by mht-sharma — COMMENTED** [review]
Thanks for the changes. I have a few comments

**Comment by mht-sharma** [comment]
Could you also document the same in docs/brevitas

File: optimum/amd/brevitas/export.py (hunk 2 above)
Comment (Giuseppe5) [reply optimum/amd/brevitas/export.py:8]: Some of the arguments for the export function have very specific values at the moment (e.g., `task` or `do_validation`, etc.), that are needed for the export to work as expected.
Should those be matched as well or left to their current values without possibility of being changed?

Comment (mht-sharma) [reply optimum/amd/brevitas/export.py:8]: We should add only the required values for this context. and set default for `do_validation` to `false` and task to `text-generation-with-past`

File: examples/quantization/brevitas/quantize_llm.py (hunk 1 above)
Comment (Giuseppe5) [reply examples/quantization/brevitas/quantize_llm.py]: I opted to keep as similar as possible to the original name so it became:
`onnx_export_from_quantized_model`

File: optimum/amd/brevitas/export.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+            custom_onnx_configs=custom_onnx_configs,
+            fn_get_submodels=fn_get_submodels,
+            _variant=_variant,
+            preprocessors=preprocessors,
+            device=device,
+            no_dynamic_axes=no_dynamic_axes,
+            use_subprocess=use_subprocess,
+            do_constant_folding=do_constant_folding,
+            task="text-generation-with-past",
+            do_validation=False,
```
Comment (mht-sharma) [inline optimum/amd/brevitas/export.py:47]:
```suggestion
            task=task,
            do_validation=False,
```

File: optimum/amd/brevitas/export.py [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    optimize: Optional[str] = None,
+    monolith: bool = False,
+    model_kwargs: Optional[Dict[str, Any]] = None,
+    custom_onnx_configs: Optional[Dict[str, "OnnxConfig"]] = None,
+    fn_get_submodels: Optional[Callable] = None,
+    _variant: str = "default",
+    preprocessors: List = None,
+    device: str = "cpu",
+    no_dynamic_axes: bool = False,
+    use_subprocess: bool = False,
```
Comment (mht-sharma) [inline optimum/amd/brevitas/export.py:26]:
```suggestion
    no_dynamic_axes: bool = False,
    task: str = "text-generation-with-past",
    use_subprocess: bool = False,
```

**Review by mht-sharma — APPROVED** [review]
LGTM!

**Pull request merged**